doc/en/developer/source/policies/pull_request.rst

```diff
@@ -0,0 +1,106 @@
+
+See the :ref:`developer quickstart <quickstart>` for information on forking and building your own copy of GeoServer from sources.
+
+Assuming that ``origin`` points to your github repo the the patch workflow then becomes:
+
+#. Make the change.::
+
+     git checkout -b my_bugfix master
+     git add .
+     git commit-m "fixed bug xyz"
+
+#. Push the change up to your github repository.::
+
+     git push origin my_bugfix
+
+#. Visit your github repo page and issue the pull request. 
+
+#. For the release notes we ask that you open a `JIRA <https://osgeo-org.atlassian.net/projects/GEOS>`_ ticket linking to your pull request.
+
+   At this point the core developers will be notified of the pull request and review it at the earliest 
+convenience.
+
+Core developers will :ref:`review` the patch and may equire changes or improvements to it prior to it being accepted. It will be up to the submitter to amend the pull request and keep it alive until it gets merged. Please be patient as pull requests are often reviewed in spare time so turn-around can be a little slow.
+
+If a pull request becomes stale with no feedback from the submitter for a couple of months long, it will linked form a JIRA issue (to avoid losing the partial work) and then be closed.
+
+Tips and Tricks
+---------------
+
+The following guidelines are meant to ensure that contirbutions submitted via pull request will be as easy as possible to review and merge.
+
+Clean Build
+^^^^^^^^^^^
+
+In general developers will only accept pull requests that apply cleanly against the latest versions of the stable and master branches. Therefore before you generate a patch it is important that you update your checkout to the latest for the branch you are working on.
+
+Ensure your IDE/editor is properly configured
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ensure that your development environment is properly configured for GeoServer development. A common issue 
+with patches from newcomers is that their text editor is configured to use tabs rather than spaces.
+
+See the the :ref:`eclipse_guide` for general information about formatting and ide setup. 
+
+Include only relevant changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ensure the patch only contains changes relevant to the issue you are trying to fix. A common mistake is 
+to include whitespace and formatting changes along with the relevant changes. These changes, while they 
+may seem harmless, make the patch much harder to read.
+
+Fix one thing at a time
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Do not batch up multiple unrelated changes into a single patch. If you want to fix multiple issues work
+on them separately and submit separate patches for them.
+
+Be patient
+^^^^^^^^^^
+
+The core developers review community patches in spare time. Be cognizant of this and realize that just 
```

Comment:
Well, the contributor might be contributing in paid time too (happens a lot), it's the unlucky reviewer that gets  nothing out of it.